### ISSUE TITLE
[bug]: Resolving Waiting Close and Unconfirmed Transactions

### ISSUE THREAD

**Issue by usernameisJim** [lightningnetwork/lnd#7993]
### Background

I currently have [this guide](https://community.umbrel.com/t/steps-for-waiting-close-channels-and-unconfirmed-transactions/12937) for Umbrel users, any feedback appreciated

Trying to follow the [LND guide ](https://community.umbrel.com/t/steps-for-waiting-close-channels-and-unconfirmed-transactions/12937)
and prepend Umbrel lncli command, 

I'm confused on which portion should be used and in each circumstance as per the LND doc,

Does bumpfee for CPFP work in most scenarios on channel closures?

and it must be this kind of command?

`sudo ~/umbrel/scripts/app compose lightning exec lnd lncli listchaintxns --start_height <current block> --end_height -1`

Or more something like this?

`sudo ~/umbrel/scripts/app compose lightning exec lnd lncli wallet bumpfee --conf_target 6 --force <txid>:<index point>`

Or can I confirm the recommended command step-by-step you should use to speed up the sat per byte on the closings showing unconfirmed? 

I'd like to explain this easier and separate what is and isn't possible, 

For waiting close and unconfirmed LND closings, should we bump fee or use the --start_height command? 

How can we tell for sure when closing between us and the peer if RBF is enabled and what to use? 

How can we tell if that user is a CLN peer?

Thanks for input

### Your environment

* version of `lnd`: LND 0.16.4-beta
* which operating system (`uname -a` on *Nix): umbrelOS v0.5.5 
* version of `btcd`, `bitcoind`, or other backend: Bitcoin Core 25.0.0
* any other relevant environment details: 
Raspberry Pi 4, (umbrelOS also runs on Debian and Ubuntu) 

### Steps to reproduce

Closing all channels in LND

### Expected behaviour

All funds returning onchain 

### Actual behaviour

Channels are stuck waiting close for longer than 1 week

**Comment by xgfreon** [comment]
Here is the pending's I currently have waiting to close. any of them with Tor address do not have a unconfirmed transaction yet in the pool. The clearnet nodes do have unconfirmed. I want to speed those up as they have been set to 10sat/vb and will never clear. 
![image](https://github.com/lightningnetwork/lnd/assets/17713661/4eb68893-5638-454f-886d-bc144f94970f)

**Comment by xgfreon** [comment]
I  have tried to bump close fee using the following command against the channelpoint of the channel open, it didnt return an error but I have not seen anything happen either. 
 ~/umbrel/scripts/app compose lightning exec lnd lncli wallet bumpclosefee --sat_per_byte 50 4a07940d0830a78e5b4535c17c89efe3454415e0c5a8288753c552d6066c72ab:1

**Comment by alexbosworth** [comment]
You don't want to bump the channel outpoint, you want to bump the anchor outpoint

**Comment by usernameisJim** [comment]
Thank you for your reply @alexbosworth, 

myself and others were having trouble with this is it true "You will only be able to make use of this command if it was created as an anchor channel. For other channels, only the counterparty of the party initiating the force close will be able to get the channel confirmed more quickly if they spend their output with a higher fee (CPFP)."

And I couldn't find example of the anchor outpoint command?

We are not able to use it in this scenario @xgfreon shared above?  What would you recommend?  Your input is really appreciated - want to learn and then help others having issues after switching setups/closures too

**Comment by alexbosworth** [comment]
The anchor output you can see is the 330 small outputs on the pending unconfirmed transaction

Bump fee can spend this outpoint into a child tx and then if that tx is a high fee it pays for the parent transaction to confirm, that's CPFP

**Comment by usernameisJim** [comment]
@alexbosworth Forgive me if I'm not yet grasping, 

I grab his anchor outpoint from the mempool, for this tx as an example? `4a07940d0830a78e5b4535c17c89efe3454415e0c5a8288753c552d6066c72ab`
https://mempool.space/tx/4a07940d0830a78e5b4535c17c89efe3454415e0c5a8288753c552d6066c72ab

and it would be 

`lncli wallet bumpclosefee --sat_per_byte 50 < ? >:<index point>`

Where does one grab that ?

tysm

**Comment by alexbosworth** [comment]
The close for that tx is `907f0ece05abcdd0ea9c7b16a3012320b3deec102ad896599e2540ab89cafa32`

The anchors on that are index 0 and 1

You can see it is already being CPFP'ed via the index 1

**Comment by usernameisJim** [comment]
:grimacing::zap: I haven't been able to find where you got this from
> `907f0ece05abcdd0ea9c7b16a3012320b3deec102ad896599e2540ab89cafa32` 


@alexbosworth

**Comment by alexbosworth** [comment]
That is the id of the unconfirmed tx that spends the channel outpoint, the commitment transaction

**Comment by xgfreon** [comment]
![image](https://github.com/lightningnetwork/lnd/assets/17713661/342cc9db-01a0-4197-9daf-b39d61580570)
 He got it from clicking on that red arrow

**Comment by usernameisJim** [comment]
Yeah my understanding is we can bump but we can't CPFP unless you also own the outputs to the TX...
so they must have CPFP'd already?

**Comment by xgfreon** [comment]
@alexbosworth if you look at another of my pending channels from above, in Lnd-g It says channel closed, however when i goto the mempool and look for a close like
 I do for the other example, I dont see a red arror next to my open channel point. How can I find the close for this channel, or maybe its been unreachable from my node and will keep trying to close on its own?
![image](https://github.com/lightningnetwork/lnd/assets/17713661/11fa6bbd-3a23-4940-b489-9afd182a16b5)

https://mempool.space/tx/fe5801b134c3b5c54a8597a630f98bf9ff68418ece0a66b9d4a2022bac598f91

I have this issue for any of my channels that were TOR only channels. I have tried to send a force close using Chantools but it tells me the address is not reachable, I have only been successful using chantools on clearnode ip's only. The node clearly shows online from amboss or 1ML.  
https://1ml.com/node/034ff2a80dcca440c34973c154792248e0df97c7650dc941c78b1258cc98578934
https://amboss.space/node/034ff2a80dcca440c34973c154792248e0df97c7650dc941c78b1258cc98578934

**Comment by usernameisJim** [comment]
@xgfreon maybe you can contact them their twitter and Telegram are linked [their](https://amboss.space/node/034ff2a80dcca440c34973c154792248e0df97c7650dc941c78b1258cc98578934) to initiate the cooperative close?

**Comment by alexbosworth** [comment]
I don't have enough info from what you posted but if you want to close a channel and you still have the channel state you can try to coop close or force close the channel

If a force close gets stuck in the mempool and it's an anchor channel and the parent is above the minimum relay fee you can fee bump the anchor

**Comment by saubyk** [comment]
Moving this issue to a discussion thread, as there is nothing to fix here.
